// saltui/lib/Slot/tpls/Header.js
'use strict';

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Context = require('../../Context');

var _Context2 = _interopRequireDefault(_Context);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SlotHeader = function (_React$Component) {
  _inherits(SlotHeader, _React$Component);

  function SlotHeader() {
    _classCallCheck(this, SlotHeader);

    return _possibleConstructorReturn(this, (SlotHeader.__proto__ || Object.getPrototypeOf(SlotHeader)).apply(this, arguments));
  }

  _createClass(SlotHeader, [{
    key: 'render',


    // http://facebook.github.io/react/docs/reusable-components.html
    value: function render() {
      var _props = this.props,
          className = _props.className,
          title = _props.title,
          confirmText = _props.confirmText,
          cancelText = _props.cancelText,
          isScrolling = _props.isScrolling,
          onConfirm = _props.onConfirm,
          onCancel = _props.onCancel,
          others = _objectWithoutProperties(_props, ['className', 'title', 'confirmText', 'cancelText', 'isScrolling', 'onConfirm', 'onCancel']);

      return _react2.default.createElement(
        'div',
        _extends({ className: _Context2.default.prefixClass('slot-header FBH FBAC') }, others),
        _react2.default.createElement(
          'div',
          { className: _Context2.default.prefixClass('slot-cancel'), onClick: onCancel },
          cancelText
        ),
        _react2.default.createElement(
          'div',
          { className: _Context2.default.prefixClass('FB1 FAC slot-title') },
          title
        ),
        _react2.default.createElement(
          'div',
          {
            className: (0, _classnames2.default)(_Context2.default.prefixClass('slot-confirm'), {
              enable: !isScrolling
            }),
            onClick: onConfirm
          },
          confirmText
        )
      );
    }
  }]);

  return SlotHeader;
}(_react2.default.Component);

SlotHeader.displayName = 'SlotHeader';
SlotHeader.propTypes = {
  title: _propTypes2.default.string,
  confirmText: _propTypes2.default.string,
  cancelText: _propTypes2.default.string,
  isScrolling: _propTypes2.default.bool,
  onConfirm: _propTypes2.default.func,
  onCancel: _propTypes2.default.func,
  className: _propTypes2.default.any
};
SlotHeader.defaultProps = {
  title: '',
  cancelText: '取消',
  confirmText: '完成',
  isScrolling: false,
  onConfirm: function onConfirm() {},
  onCancel: function onCancel() {},

  className: undefined
};


module.exports = SlotHeader;
